### PR TITLE
Add `all` on projection #300

### PR DESCRIPTION
**Pull request by jxnu-liguobin**

### CODE REVIEW

File: docs/client-side-cases-by-proxy.md [hunk 1]
```diff
@@ -9,11 +9,12 @@ However, since we have simplified the client-side logic as much as possible, thi
 so that the users do not need to choose the fields on projection. 
 This will also have a bad effect, we can only use all fields of the projection on the returned structure by default. 
 Moreover, we also need set to default the depth of nested queries so that the proxy can end smoothly. 
-All in all, the scenarios used are limited. Currently `.. on` and `alias` are not supported.
+All in all, the scenarios used are limited. Currently, `.. on` and `alias` not be supported.
```
Comment (joffrey-bion) [inline docs/client-side-cases-by-proxy.md:12]: Why this change? *"X and Y are not supported"* was grammatically correct, *"X and Y not be supported"* is not correct.
I agree with the added comma, but the verb needs to stay conjugated.

Comment (jxnu-liguobin) [reply docs/client-side-cases-by-proxy.md:12]: I want to say that this is passive

Comment (joffrey-bion) [reply docs/client-side-cases-by-proxy.md:12]: But this *is* the passive voice. The active voice would be *"we don't support X"*. The passive voice is *"X is not supported"*. Using "be" here is just incorrect.

Comment (jxnu-liguobin) [reply docs/client-side-cases-by-proxy.md]: ok,thanks.

File: src/main/java/com/kobylynskyi/graphql/codegen/model/graphql/GraphQLResponseProjection.java [hunk 2]
```diff
@@ -12,6 +10,36 @@
...
+     * }}
+     * Map Notes:
+     * `key` is parentProjection.childProjection.currentMethod. e.g. `CharacterResponseProjection.CharacterResponseProjection.friends` (excluding the first layer, so if only want the first child layer, use `all$(1)`)
+     * `value` is current depth for Character type. Each projection has a new instance of `projectionDepthOnFields`, so it always be `1` or `0`.
+     * and `responseProjectionMaxDepth` will reduce by recursive.
+     */
+    protected final Map<String, Integer> projectionDepthOnFields = new HashMap<>();
+
+    //Defined at the parent level to use dynamic calls, default null.
+    public GraphQLResponseProjection all$() {
```
Comment (kobylynskyi) [inline src/main/java/com/kobylynskyi/graphql/codegen/model/graphql/GraphQLResponseProjection.java:35]: I think we should use:
`public abstract GraphQLResponseProjection all$();`
`public abstract GraphQLResponseProjection all$(int maxDepth);`

**Review by kobylynskyi — CHANGES_REQUESTED** [review]

File: src/main/java/com/kobylynskyi/graphql/codegen/model/graphql/GraphQLResponseProjection.java (hunk 2 above)
Comment (jxnu-liguobin) [reply src/main/java/com/kobylynskyi/graphql/codegen/model/graphql/GraphQLResponseProjection.java:35]: OK😀

**Review by kobylynskyi — APPROVED** [review]
